pkg/cmd/adm/register_member_test.go

```diff
@@ -2,215 +2,622 @@ package adm
+					if hostReady {
+						status = corev1.ConditionTrue
+					}
+					// there's always at most 1 toolchain cluster for the host in the member cluster
+					expectedHostToolchainClusterName, err := utils.GetToolchainClusterName("host", "https://cool-server.com", "")
+					require.NoError(t, err)
+					expectedHostToolchainCluster := &toolchainv1alpha1.ToolchainCluster{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      expectedHostToolchainClusterName,
+							Namespace: "toolchain-member-operator",
+						},
+						Spec: toolchainv1alpha1.ToolchainClusterSpec{
+							APIEndpoint: "https://cool-server.com",
+						},
+						Status: toolchainv1alpha1.ToolchainClusterStatus{
+							Conditions: []toolchainv1alpha1.ToolchainClusterCondition{
+								{
+									Type:   toolchainv1alpha1.ToolchainClusterReady,
+									Status: status,
+								},
+							},
+						},
+					}
+					persist(expectedHostToolchainCluster)
+				} else {
+					AssertFirstArgPrefixRestEqual("(.*)/add-cluster-(.*)", expectedMemberArgs...)(t, args...)
+					status := corev1.ConditionFalse
+					if memberReady {
+						status = corev1.ConditionTrue
+					}
+					expectedMemberToolchainClusterName, err := utils.GetToolchainClusterName("member", "https://cool-server.com", nameSuffix)
+					require.NoError(t, err)
+					expectedMemberToolchainCluster := &toolchainv1alpha1.ToolchainCluster{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      expectedMemberToolchainClusterName,
+							Namespace: "toolchain-host-operator",
+							Labels: map[string]string{
+								"namespace": "toolchain-member-operator",
+							},
+						},
+						Spec: toolchainv1alpha1.ToolchainClusterSpec{
+							APIEndpoint: "https://cool-server.com",
+						},
+						Status: toolchainv1alpha1.ToolchainClusterStatus{
+							Conditions: []toolchainv1alpha1.ToolchainClusterCondition{
+								{
+									Type:   toolchainv1alpha1.ToolchainClusterReady,
+									Status: status,
+								},
+							},
+						},
+					}
+					persist(expectedMemberToolchainCluster)
+				}
+				counter++
+			})
+	}
+
+	t.Run("produces valid example SPC", func(t *testing.T) {
+		// given
 		term := NewFakeTerminalWithResponse("Y")
-		toolchainConfig := config.NewToolchainConfigObj(t, config.AutomaticApproval().Enabled(true))
-		newClient, fakeClient := NewFakeClients(t, toolchainConfig, deployment)
-		numberOfUpdateCalls := 0
-		fakeClient.MockUpdate = whenDeploymentThenUpdated(t, fakeClient, hostDeploymentName, 1, &numberOfUpdateCalls)
-		ctx := clicontext.NewCommandContext(term, newClient)
+		newClient, fakeClient := newFakeClientsFromRestConfig(t, deployment)
+		ctx := newExtendedCommandContext(term, newClient)
 		counter = 0
+		addClusterCommand := commandCreator(fakeClient, true, true, "", false)
+		expectedHostArgs = []string{"--type", "host", "--host-kubeconfig", hostKubeconfig, "--host-ns", "toolchain-host-operator", "--member-kubeconfig", memberKubeconfig, "--member-ns", "toolchain-member-operator"}
```

Comment:
minor and partially related to this PR - If I'm not mistaken those are "global variables" , it's not obvious where those are used/incremented at first, maybe passing those as parameters would make it cleaner. Feel free to ignore this for now, we can change it later if we find it useful.

Reply:
I'm referring to the fact that `counter` is defined in the test but is not clear where/when it gets incremented at first. Also the `expected*Args` list is defined here but not clear where is needed.

Reply:
👍🏼 great observation. I tried to redo it.

Reply:
nice, thanks for changing this. The new setup struct looks great!